checks/evaluation/branch_protection_test.go

```diff
@@ -18,563 +18,1691 @@ import (
 	"testing"
 
 	"github.com/ossf/scorecard/v4/checker"
-	"github.com/ossf/scorecard/v4/clients"
+	sce "github.com/ossf/scorecard/v4/errors"
+	"github.com/ossf/scorecard/v4/finding"
 	scut "github.com/ossf/scorecard/v4/utests"
 )
 
-func testScore(branch *clients.BranchRef, codeownersFiles []string, dl checker.DetailLogger) (int, error) {
-	var score levelScore
-	score.scores.basic, score.maxes.basic = basicNonAdminProtection(branch, dl)
-	score.scores.review, score.maxes.review = nonAdminReviewProtection(branch)
-	score.scores.adminReview, score.maxes.adminReview = adminReviewProtection(branch, dl)
-	score.scores.context, score.maxes.context = nonAdminContextProtection(branch, dl)
-	score.scores.thoroughReview, score.maxes.thoroughReview = nonAdminThoroughReviewProtection(branch, dl)
-	score.scores.adminThoroughReview, score.maxes.adminThoroughReview = adminThoroughReviewProtection(branch, dl)
-	score.scores.codeownerReview, score.maxes.codeownerReview = codeownerBranchProtection(branch, codeownersFiles, dl)
-
-	return computeFinalScore([]levelScore{score})
-}
-
-// TODO: order of tests to have progressive scores.
-func TestIsBranchProtected(t *testing.T) {
+func TestBranchProtection(t *testing.T) {
 	t.Parallel()
-	trueVal := true
-	falseVal := false
-	var zeroVal int32
-	var oneVal int32 = 1
-	branchVal := "branch-name"
 	tests := []struct {
-		name            string
-		branch          *clients.BranchRef
-		codeownersFiles []string
-		expected        scut.TestReturn
+		name     string
+		findings []finding.Finding
+		result   scut.TestReturn
 	}{
 		{
-			name: "GitHub default settings",
-			expected: scut.TestReturn{
-				Error:         nil,
-				Score:         3,
-				NumberOfWarn:  6,
-				NumberOfInfo:  2,
-				NumberOfDebug: 1,
-			},
-			branch: &clients.BranchRef{
-				Name:      &branchVal,
-				Protected: &trueVal,
-				BranchProtectionRule: clients.BranchProtectionRule{
-					AllowDeletions:          &falseVal,
-					AllowForcePushes:        &falseVal,
-					RequireLinearHistory:    &falseVal,
-					EnforceAdmins:           &falseVal,
-					RequireLastPushApproval: &falseVal,
-					RequiredPullRequestReviews: clients.PullRequestReviewRule{
-						Required: &falseVal,
+			name: "Required status check enabled",
```

Comment:
the name of this test is "Required status check enabled", but the test result is an error / inconclusive score due to the missing branch names. Something seems like a mismatch here.

Reply:
Thanks! Renamed in https://github.com/ossf/scorecard/pull/3759/commits/ca2eae69d009ceef6c3e1f063d508e8059bb4bec